### PR TITLE
Possibility of passing ConstraintValidatorFactory to SymfonyValidator

### PR DESCRIPTION
**Pull request by martenb**
It would be nice, to have possibility of passing custom ConstraintValidatorFactory to SymfonyValidator like this

https://github.com/contributte/validator/blob/master/src/ContainerConstraintValidatorFactory.php

to be able to use DI injections in custom validators.

### CODE REVIEW

**Comment by martenb** [comment]
Another option is to completely change the interface of SymfonyValidator to

```php
public function __construct(\Symfony\Component\Validator\Validator\ValidatorInterface $validator)
```
whitch register [contributte/validator](https://github.com/contributte/validator), so we don't need nettrine/annotations anymore...

File: src/Mapping/Validator/SymfonyValidator.php [hunk 1]
```diff
@@ -14,9 +15,13 @@ class SymfonyValidator implements IEntityValidator
 	/** @var Reader */
 	private $reader;
 
-	public function __construct(Reader $reader)
+	/** @var ConstraintValidatorFactoryInterface|null */
+	private $constraintValidatorFactory;
+
+	public function __construct(Reader $reader, ?ConstraintValidatorFactoryInterface $constraintValidatorFactory = null)
```
Comment (f3l1x) [inline src/Mapping/Validator/SymfonyValidator.php:21]: I like it, can you please change it to setter?

Comment (martenb) [reply src/Mapping/Validator/SymfonyValidator.php:21]: Setter is not a problem, but i do not know, how to correctly write it i neon,, i am trying somenting like this

```yaml
api:
    plugins:
        Apitte\Core\DI\Plugin\CoreMappingPlugin:
            request:
                validator:
                    factory: Apitte\Core\Mapping\Validator\SymfonyValidator(@phpdoc.reader)
                    setup:
                        - setConstraintValidatorFactory(Contributte\Validator\ContainerConstraintValidatorFactory())
```
And i am getting `Nette\InvalidArgumentException: Argument is not valid Statement entity.`

Comment (martenb) [reply src/Mapping/Validator/SymfonyValidator.php:21]: It must be defined as service and then passed to validator?

```yaml
api:
    plugins:
        Apitte\Core\DI\Plugin\CoreMappingPlugin:
            request:
                validator: @Apitte\Core\Mapping\Validator\SymfonyValidator

services:
    -
        factory: Apitte\Core\Mapping\Validator\SymfonyValidator(@phpdoc.reader)
        setup:
            - setConstraintValidatorFactory(Contributte\Validator\ContainerConstraintValidatorFactory())
```

**Comment by martenb** [comment]
> Another option is to completely change the interface of SymfonyValidator to
> 
> ```
> public function __construct(\Symfony\Component\Validator\Validator\ValidatorInterface $validator)
> ```
> 
> whitch register [contributte/validator](https://github.com/contributte/validator), so we don't need nettrine/annotations anymore...

**Comment by martenb** [comment]
What do you think? Chainable setter looks better in config...?

**Comment by f3l1x** [comment]
I have merge it. Thanks. :-) 

https://github.com/apitte/core/commit/4cbc1c072630a91b57f3e9be83ea99680ab37e0e

**Pull request closed without merging**